doc/devel/min_dep_policy.rst

```diff
@@ -0,0 +1,38 @@
+.. _min_deps_policy:
+
+======================================
+Minimum Version Of dependencies Policy
+======================================
+
+Python and numpy
+================
+
+- support minor versions of ``Python`` initially released in the previous
+  3 years
+- support minor versions of ``numpy`` initially released in the
+  previous 3 years or oldest that supports the minimum python version
+  (which ever is higher)
+
+We will bump the minimum python and numpy versions as we can every
+minor and major release, but never on a patch release.
+
+Other Dependencies
```

Comment:
I would split this into separate sections "Python packages" and "System/c-dependencies".

Reply:
I like that.